### PR TITLE
Revise cos

### PR DESCRIPTION
**Pull request by pszmel**
### Details:
 - update docs
 - add RTTI macro
 - create reference impl for int input
 - create visitor test
 - create backend test for int input
 - create type_prop tests

### Tickets:
 - 37431

### CODE REVIEW

File: docs/ops/arithmetic/Cos_1.md [hunk 1]
```diff
@@ -2,31 +2,31 @@
...
 
 * *T*: any numeric type.
 
-*Cos* does the following with the input tensor *a*:
-
-\f[
-a_{i} = cos(a_{i})
-\f]
 
 **Examples**
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Cos_1.md:31]: You could maybe remove \*Example 1\* and change \*\*Examples\*\* to ---> \*\*Example\*\*

File: docs/ops/arithmetic/Cos_1.md [hunk 2]
```diff
@@ -2,31 +2,31 @@
...
 
 **Inputs**
 
-* **1**: An tensor of type T. **Required.**
+* **1**: A tensor of type *T* and arbitrary shape. **Required.**
 
 **Outputs**
 
-* **1**: The result of element-wise cos operation. A tensor of type T.
+* **1**: The result of element-wise cos operation. A tensor of type *T*.
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Cos_1.md:24]: The result of element-wise cos operation ---> The result of element-wise \*Cos\* operation
A tensor of type \*T\* ---> A tensor of type \*T\* and the same shape as the input tensor

File: docs/ops/arithmetic/Cos_1.md [hunk 3]
```diff
@@ -2,31 +2,31 @@
 
 **Versioned name**: *Cos-1*
 
-**Category**: Arithmetic unary operation 
+**Category**: Arithmetic unary operation
 
 **Short description**: *Cos* performs element-wise cosine operation with given tensor.
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Cos_1.md:7]: *Cos* performs element-wise cosine operation on a given input tensor.

File: docs/ops/arithmetic/Cos_1.md [hunk 4]
```diff
@@ -2,31 +2,31 @@
...
 
-**Attributes**:
+**Detailed description**: For each element from the input tensor calculates corresponding
+element in the output tensor with the following formula:
```
Comment (ggalieroc) [inline docs/ops/arithmetic/Cos_1.md:10]: \*Cos\* performs element-wise cosine operation on a given input tensor, based on the following mathematical formula:

**Review by ggalieroc — APPROVED** [review]

**Comment by ggalieroc** [comment]
As changes required for revising Cos op are small, this PR comprises all sub-tasks:
@lazarevevgeny, could you please review the spec part?
@iefode, SLTs with int precision have been added + op added to trusted op list

cc @bszmelcz this PR is similar to the one you are currently working in. Please have a look

**Review by bszmelcz — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]
@lazarevevgeny please review the specification

**Comment by ggalieroc** [comment]
Also add checks for inputs/outputs in [evaluate ](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/op/cos.cpp#L69) method, similarly as in [here](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/op/prelu.cpp#L81)

**Comment by pszmel** [comment]
@ggalieroc  added check.

File: ngraph/core/src/op/cos.cpp [hunk 5]
```diff
@@ -69,6 +69,7 @@ namespace cosop
 bool op::Cos::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const
 {
     NGRAPH_OP_SCOPE(v0_Cos_evaluate);
+    NGRAPH_CHECK(validate_host_tensor_vector(outputs, 1) && validate_host_tensor_vector(inputs, 2));
```
Comment (ggalieroc) [inline ngraph/core/src/op/cos.cpp:72]: The operation is unary, it only has one input. This check should be:
`NGRAPH_CHECK(validate_host_tensor_vector(outputs, 1) && validate_host_tensor_vector(inputs, 1));`

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**